lib/Client.js

```diff
@@ -7,9 +7,22 @@ var min = require('osc-min');
 var Message = require('./Message');
 
 var Client = function (host, port) {
+    var isBroadcast = host.split('.').reduce(function(previous, current){
+        return previous ? true : +current === 255;
+    }, false),
+    setter = function () {
+        this.setBroadcast(isBroadcast);
+    };
+
     this.host = host;
     this.port = port;
     this._sock = dgram.createSocket('udp4');
+
+    // check if the host is a broadcast address
+    // then switch to broadcast mode in callback
+    //
+    this._sock.bind(setter.bind(this._sock));
```

Comment:
I'm in a double bind here... Actually one of them is a bind to the port, the second one is a bind to the function. So not as bad, but I agree that it looks confusing.

I will get back to you in a bit.